bootstrapper/cmd/bootstrapper/run.go

```diff
@@ -67,12 +70,26 @@ func run(issuer atls.Issuer, openDevice vtpm.TPMOpenFunc, fileHandler file.Handl
 	go cleaner.Start()
 	defer cleaner.Done()
 
-	joinClient.Start(cleaner)
+	var wg sync.WaitGroup
 
-	if err := initServer.Serve(bindIP, bindPort, cleaner); err != nil {
-		log.With(slog.Any("error", err)).Error("Failed to serve init server")
-		os.Exit(1)
-	}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if err := joinClient.Start(cleaner); err != nil {
+			log.With(slog.Any("error", err)).Error("Failed to join cluster")
+			reboot(fmt.Errorf("joining cluster: %w", err))
+		}
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if err := initServer.Serve(bindIP, bindPort, cleaner); err != nil {
+			log.With(slog.Any("error", err)).Error("Failed to serve init server")
+			reboot(fmt.Errorf("serving init server: %w", err))
+		}
+	}()
+	wg.Wait()
```

Comment:
It seems like both initserver and joinclient now have internal logic for differentiating non-recoverable errors. Would it make sense to check the error type here in run instead so that we only have one place that deals with disk reset?

Reply:
Decided to move the disk wiping into the `run` function in case `InitServer` or `JoinClient` return an error.
We can do this unconditionally in this case, as we are sure the node was not previously bootstrapped, and setting the token to `false` is safe to do, regardless if it was set to `true` or not at all.
This takes out a bit of complexity when it comes to determining failure states.